bug: tests failing after NVD server update

There are currently 20 tests failing (details below).  Since we're a few days after NVD has done their server move and updates, I strongly suspect that a data change has caused these and all these tests will need to be updated with new exptected data, or that our cache got somehow corrupted and will need to be refreshed.  I haven't actually investigated them yet and will likely disable them temporarily while we debug.

```python
=========================== short test summary info ============================
FAILED test/test_csv2cve.py::TestCSV2CVE::test_csv2cve_valid_file - AssertionError: assert ('cve_bin_tool', 20, 'There are 2 products with known CVEs detected') in [('cve_bin_tool', 20, 'CVE Binary Tool v3.3'), ('cve_bin_tool', 20, 'This product uses the NVD API but is not endorsed or certified by the NVD.'), ('cve_bin_tool.CVEDB', 20, 'Using cached CVE data (<24h old). Use -u now to update immediately.'), ('cve_bin_tool.CVEDB', 20, 'There are 17109 CVE entries in the database'), ('cve_bin_tool.CVEDB', 20, 'There are 17109 CVE entries from REDHAT in the database'), ('cve_bin_tool', 20, 'CVE database contains CVEs from National Vulnerability Database (NVD), Open Source Vulnerability Database (OSV), Gitlab Advisory Database (GAD) and RedHat'), ...]
 +  where [('cve_bin_tool', 20, 'CVE Binary Tool v3.3'), ('cve_bin_tool', 20, 'This product uses the NVD API but is not endorsed or certified by the NVD.'), ('cve_bin_tool.CVEDB', 20, 'Using cached CVE data (<24h old). Use -u now to update immediately.'), ('cve_bin_tool.CVEDB', 20, 'There are 17109 CVE entries in the database'), ('cve_bin_tool.CVEDB', 20, 'There are 17109 CVE entries from REDHAT in the database'), ('cve_bin_tool', 20, 'CVE database contains CVEs from National Vulnerability Database (NVD), Open Source Vulnerability Database (OSV), Gitlab Advisory Database (GAD) and RedHat'), ...] = <_pytest.logging.LogCaptureFixture object at 0x7ff4abb82c50>.record_tuples
FAILED test/test_exploits.py::TestExploitScanner::test_exploit_checker[True-exploits_list0-product_info0-triage_info0-CRITICAL-EXPLOIT] - IndexError: list index out of range
FAILED test/test_exploits.py::TestExploitScanner::test_exploit_checker[False-exploits_list1-product_info1-triage_info1-CRITICAL] - IndexError: list index out of range
FAILED test/test_exploits.py::TestExploitScanner::test_exploit_checker[True-exploits_list2-product_info2-triage_info2-CRITICAL] - IndexError: list index out of range
FAILED test/test_exploits.py::TestExploitScanner::test_exploit_checker[False-exploits_list3-product_info3-triage_info3-CRITICAL] - IndexError: list index out of range
FAILED test/test_language_scanner.py::TestLanguageScanner::test_java_package[/home/runner/work/cve-bin-tool/cve-bin-tool/test/language_data/pom.xml-product_list0] - UnboundLocalError: cannot access local variable 'product_info' where it is not associated with a value
FAILED test/test_language_scanner.py::TestLanguageScanner::test_python_package[/home/runner/work/cve-bin-tool/cve-bin-tool/test/language_data/PKG-INFO] - UnboundLocalError: cannot access local variable 'product_info' where it is not associated with a value
FAILED test/test_sbom.py::TestSBOM::test_valid_spdx_file[/home/runner/work/cve-bin-tool/cve-bin-tool/test/sbom/spdx_test.spdx-spdx_parsed_data0] - AssertionError: assert ProductInfo(vendor='gnu', product='glibc', version='2.11.1') in defaultdict(<class 'dict'>, {ProductInfo(vendor='UNKNOWN', product='glibc', version='2.11.1'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='apache-jena', version='3.12.0'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='saxon', version='8.8'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}})
FAILED test/test_sbom.py::TestSBOM::test_valid_spdx_file[/home/runner/work/cve-bin-tool/cve-bin-tool/test/sbom/spdx_test.spdx.rdf-spdx_parsed_data1] - AssertionError: assert ProductInfo(vendor='gnu', product='glibc', version='2.11.1') in defaultdict(<class 'dict'>, {ProductInfo(vendor='UNKNOWN', product='glibc', version='2.11.1'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='saxon', version='8.8'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='jena', version='3.12.0'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}})
FAILED test/test_sbom.py::TestSBOM::test_valid_spdx_file[/home/runner/work/cve-bin-tool/cve-bin-tool/test/sbom/spdx_test.spdx.json-spdx_parsed_data2] - AssertionError: assert ProductInfo(vendor='gnu', product='glibc', version='2.11.1') in defaultdict(<class 'dict'>, {ProductInfo(vendor='UNKNOWN', product='glibc', version='2.11.1'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='apache-jena', version='3.12.0'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='saxon', version='8.8'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}})
FAILED test/test_sbom.py::TestSBOM::test_valid_spdx_file[/home/runner/work/cve-bin-tool/cve-bin-tool/test/sbom/spdx_test.spdx.xml-spdx_parsed_data3] - AssertionError: assert ProductInfo(vendor='gnu', product='glibc', version='2.11.1') in defaultdict(<class 'dict'>, {ProductInfo(vendor='UNKNOWN', product='glibc', version='2.11.1'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='jena', version='3.12.0'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='saxon', version='8.8'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}})
FAILED test/test_sbom.py::TestSBOM::test_valid_spdx_file[/home/runner/work/cve-bin-tool/cve-bin-tool/test/sbom/spdx_test.spdx.yml-spdx_parsed_data4] - AssertionError: assert ProductInfo(vendor='gnu', product='glibc', version='2.11.1') in defaultdict(<class 'dict'>, {ProductInfo(vendor='UNKNOWN', product='glibc', version='2.11.1'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='apache-jena', version='3.12.0'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='saxon', version='8.8'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}})
FAILED test/test_sbom.py::TestSBOM::test_valid_spdx_file[/home/runner/work/cve-bin-tool/cve-bin-tool/test/sbom/spdx_test.spdx.yaml-spdx_parsed_data5] - AssertionError: assert ProductInfo(vendor='gnu', product='glibc', version='2.11.1') in defaultdict(<class 'dict'>, {ProductInfo(vendor='UNKNOWN', product='glibc', version='2.11.1'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='apache-jena', version='3.12.0'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='saxon', version='8.8'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}})
FAILED test/test_sbom.py::TestSBOM::test_valid_spdx_file[/home/runner/work/cve-bin-tool/cve-bin-tool/test/sbom/spdx_mixed_test.spdx.json-spdx_parsed_data6] - AssertionError: assert ProductInfo(vendor='gnu', product='glibc', version='2.11.1') in defaultdict(<class 'dict'>, {ProductInfo(vendor='UNKNOWN', product='glibc', version='2.11.1'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='apache-jena', version='3.12.0'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='saxon', version='8.8'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}})
FAILED test/test_sbom.py::TestSBOM::test_valid_cyclonedx_file[/home/runner/work/cve-bin-tool/cve-bin-tool/test/sbom/cyclonedx_test.xml-cyclonedx_parsed_data0] - AssertionError: assert ProductInfo(vendor='gnu', product='glibc', version='2.11.1') in defaultdict(<class 'dict'>, {ProductInfo(vendor='UNKNOWN', product='acme application', version='9.1.1'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='tomcat-catalina', version='9.0.14'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='glibc', version='2.11.1'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}})
FAILED test/test_sbom.py::TestSBOM::test_valid_cyclonedx_file[/home/runner/work/cve-bin-tool/cve-bin-tool/test/sbom/cyclonedx_test.json-cyclonedx_parsed_data1] - AssertionError: assert ProductInfo(vendor='gnu', product='glibc', version='2.11.1') in defaultdict(<class 'dict'>, {ProductInfo(vendor='UNKNOWN', product='acme application', version='9.1.1'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='tomcat-catalina', version='9.0.14'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='glibc', version='2.11.1'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}})
FAILED test/test_sbom.py::TestSBOM::test_valid_cyclonedx_file[/home/runner/work/cve-bin-tool/cve-bin-tool/test/sbom/cyclonedx_test2.json-cyclonedx_parsed_data2] - AssertionError: assert ProductInfo(vendor='ubuntu', product='ubuntu', version='22.04') in defaultdict(<class 'dict'>, {ProductInfo(vendor='UNKNOWN', product='acme application', version='9.1.1'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='tomcat-catalina', version='9.0.14'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='ubuntu', version='22.04'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}})
FAILED test/test_sbom.py::TestSBOM::test_valid_cyclonedx_file[/home/runner/work/cve-bin-tool/cve-bin-tool/test/sbom/cyclonedx_mixed_test.json-cyclonedx_parsed_data3] - AssertionError: assert ProductInfo(vendor='gnu', product='glibc', version='2.11.1') in defaultdict(<class 'dict'>, {ProductInfo(vendor='UNKNOWN', product='acme application', version='9.1.1'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='tomcat-catalina', version='9.0.14'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='glibc', version='2.11.1'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='invalid_purl_package', version='1.1.0'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}})
FAILED test/test_sbom.py::TestSBOM::test_valid_swid_file[/home/runner/work/cve-bin-tool/cve-bin-tool/test/sbom/swid_test.xml-swid_parsed_data0] - AssertionError: assert ProductInfo(vendor='gnu', product='glibc', version='2.11.1') in defaultdict(<class 'dict'>, {ProductInfo(vendor='UNKNOWN', product='windows embedded standard 7', version='6.1.7601'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='windows embedded standard 7 with sp1 patches', version='3.0'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='sql 2005 express', version='9.00.5000.00,SP4'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='.net frame work', version='2.1.21022.8,SP2'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='glibc', version='2.11.1'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='java 8', version='1.8'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='tomcat 9', version='9.037'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}, ProductInfo(vendor='UNKNOWN', product='spring framework', version='4.7'): {'default': {'remarks': <Remarks.NewFound: 1>, 'comments': '', 'severity': ''}, 'paths': {''}}})
FAILED test/test_triage.py::test_triage - assert 0 >= 1
 +  where 0 = len([])
=========== 20 failed, 1905 passed, 26 skipped in 915.56s (0:15:15) ============
```

--- a/cve_bin_tool/cvedb.py
+++ b/cve_bin_tool/cvedb.py
@@ -54,10 +54,10 @@ class CVEDB:
     BACKUPCACHEDIR = DISK_LOCATION_BACKUP
     LOGGER = LOGGER.getChild("CVEDB")
     SOURCES = [
-        nvd_source.NVD_Source,
         curl_source.Curl_Source,
         osv_source.OSV_Source,
         gad_source.GAD_Source,
+        nvd_source.NVD_Source,  # last to avoid data overwrites
     ]
 
     INSERT_QUERIES = {

--- a/test/test_csv2cve.py
+++ b/test/test_csv2cve.py
@@ -11,7 +11,6 @@ from cve_bin_tool.error_handler import ERROR_CODES, InsufficientArgs
 
 
 class TestCSV2CVE:
-    @pytest.mark.skip(reason="Temporarily disabled -- may need data changes")
     @pytest.mark.asyncio
     async def test_csv2cve_valid_file(self, caplog):
         file_path = join(dirname(__file__), "csv", "triage.csv")

--- a/test/test_exploits.py
+++ b/test/test_exploits.py
@@ -8,7 +8,6 @@ from cve_bin_tool.util import ProductInfo
 
 
 class TestExploitScanner:
-    @pytest.mark.skip(reason="Temporarily disabled -- may need data changes")
     @pytest.mark.parametrize(
         "check_exploits, exploits_list, product_info, triage_info, expected_result",
         (

--- a/test/test_language_scanner.py
+++ b/test/test_language_scanner.py
@@ -163,7 +163,6 @@ class TestLanguageScanner:
         cls.cvedb.get_cvelist_if_stale()
         print("Database setup complete.")
 
-    @pytest.mark.skip(reason="Temporarily disabled -- may need data changes")
     @pytest.mark.parametrize(
         "filename, product_list",
         (((str(TEST_FILE_PATH / "pom.xml")), ["jmeter", "hamcrest"]),),
@@ -235,7 +234,6 @@ class TestLanguageScanner:
             assert p in found_product
         assert file_path == filename
 
-    @pytest.mark.skip(reason="Temporarily disabled -- may need data changes")
     @pytest.mark.parametrize("filename", ((str(TEST_FILE_PATH / "PKG-INFO")),))
     def test_python_package(self, filename: str) -> None:
         scanner = VersionScanner()

--- a/test/test_sbom.py
+++ b/test/test_sbom.py
@@ -68,7 +68,6 @@ class TestSBOM:
         sbom_engine = SBOMManager(filename, sbom_type)
         assert sbom_engine.scan_file() == {}
 
-    @pytest.mark.skip(reason="Temporarily disabled -- may need data changes")
     @pytest.mark.parametrize(
         "filename, spdx_parsed_data",
         (
@@ -89,7 +88,6 @@ class TestSBOM:
         for p in spdx_parsed_data:
             assert p in scan_result
 
-    @pytest.mark.skip(reason="Temporarily disabled -- may need data changes")
     @pytest.mark.parametrize(
         "filename, cyclonedx_parsed_data",
         (
@@ -107,7 +105,6 @@ class TestSBOM:
         for p in cyclonedx_parsed_data:
             assert p in scan_result
 
-    @pytest.mark.skip(reason="Temporarily disabled -- may need data changes")
     @pytest.mark.parametrize(
         "filename, swid_parsed_data",
         ((str(SBOM_PATH / "swid_test.xml"), PARSED_SBOM_DATA),),

